### ISSUE TITLE
CRASH SIGILL signal error with running DynamoRIO client on Fedora 40 and Ubuntu 24.04 versions

### ISSUE THREAD

**Issue by Mr-Grag-Universe** [DynamoRIO/dynamorio#7008]
# Describe the bug

Currently I develop a custom fuzzer based on LibFuzzer using DynamoRIO for coverage extantion. Before this bug used Fedora Linux system 38-39. This Summer I moved to the Kubuntu and tried to run my project there, but I faced the bug.
My client just cannot start and crashed before any commands in main function.
![image](https://github.com/user-attachments/assets/318f0055-d4ab-42af-b422-7388ae36a22e)

I run my program under the GDB and here is the result:
![image](https://github.com/user-attachments/assets/f9abc5d6-306a-4003-bdf5-ccf2ee92ce1a)

# To Reproduce
I run command 
``` bash
./DynamoRIO_Linux_10/bin64/drrun -c ./bin/libclient.so -- ./bin/fuzz_app
```
on Fedora 40 and Kubuntu 24.04.1 where it does not works.
As well as with command

``` bash
./DynamoRIO_Linux_10/bin64/drrun -c ./bin/libclient.so -- ls
```

Without client it works fine:
``` bash
./DynamoRIO_Linux_10/bin64/drrun ./bin/fuzz_app
```
or 
``` bash
./DynamoRIO_Linux_10/bin64/drrun ls
```

There is no difference when I run client with debug option.

So I guess it's only DR client running problem. As I discovered some bad things happening when DR tries to handle the threads to tun client.

# Expected behavior
here is a screenshot of correct fuzzer work:
![image](https://github.com/user-attachments/assets/b9fa27f4-b85d-4581-a90a-97043cd09c65)

# Versions
I use the latest release of DynamoRIO from here https://dynamorio.org/page_releases.html
I work on such systems:
Fedora 40 (in container)
![image](https://github.com/user-attachments/assets/69cf7072-efb3-4840-814f-8f565a9caeee)
Kubuntu 24.04.1 (main system)
![image](https://github.com/user-attachments/assets/8c1445db-a80c-4adf-a86b-2b2496ad69bb)

**Comment by derekbruening** [comment]
As documented, DR sends itself a SIGILL and handles it, so there is no crash involving SIGILL: https://dynamorio.org/page_debugging.html#autotoc_md142

As for the floating-point fault, that is almost certainly due to recent glibc bad designs with hidden dependencies: #5437.  Probably the latest weekly build at 
https://github.com/DynamoRIO/dynamorio/releases fixes it; if not please update #5437.

Often it is a good idea to first ask questions on the users list.

Closing as a duplicate of #5437.

**Comment by Mr-Grag-Universe** [comment]
Thanks. I've seen this issue before asked this question, but I guessed, that my problem was not related to this question closely enough. I also apologize, if this issue seems garbagelike)
